### ISSUE TITLE
Add a "Very Unsafe Save" option

### ISSUE THREAD

**Issue by droidmonkey** [keepassxreboot/keepassxc#6335]
## Summary
[TIP]:  # ( DO NOT include screenshots of your actual database! )
[NOTE]: # ( Provide a brief overview of what the new feature is all about )
Add an option to allow overwriting the existing database file directly instead of deleting the current file and renaming/moving the new one in place. This will help solve issues with some cloud storage services and with GVFS on Linux.

## Examples
[NOTE]: # ( Show us a picture or mock-up of your proposal )

## Context
[NOTE]: # ( Why does this feature matter to you? What unique circumstances do you have? )

**Comment by kuntergunt** [comment]
I don't think this is an "unsafe" operation, this is the usual way to work with files. 
You open a file for editing in write mode, this means you also lock it. Then you do your changes and save them, because the file is locked, in a save way to the file. Finally you close the file. 
Just think of doing it your way for a multi TB database. But I think I know what you have in mind: if the file is distributed (GVFS) you don't want it to have an inconsistent state on the other side. But can't you save it in a consistent way?

**Comment by droidmonkey** [comment]
It is unsafe from the aspect of protection from interruption due to crashes, power failure, etc. In direct write mode you necessarily truncate the file first (basically set the file end equal to file start) then begin writing. If it fails in the middle then you will have destroyed your database file. A safe save writes the file separately then performs an atomic "move" operation that ensures the written data is still in tact and not susceptible to interruption failure.

Keepass databases are not meant to store terabytes or even really megabytes worth of data. They are not structured and the encryption is not handled in such a way that you can modify only a small part of the file at a time. It's all or nothing.

**Comment by phoerious** [comment]
Atomic moves are not inherently incompatible with large files. Basically every application does it, including Photoshop.

**Comment by rattkin** [comment]
it would be nice if the "backup database file before saving" would work. that way it would be pretty safe
